schemes/rayleigh_friction/rayleigh_friction.F90

```diff
@@ -0,0 +1,160 @@
+
+! We apply a profile of the form otau0 * [1 + tanh (x)] / 2 , where
+! x = (k0 - k) / krange. The default is for x to equal 2 at k=1, meaning
+! krange = (k0 - 1) / 2. The default is applied when raykrange is set to 0.
+! If otau0 = 0, no term is applied.
+
+contains
+
+  !> \section arg_table_rayleigh_friction_init Argument Table
+  !! \htmlinclude rayleigh_friction_init.html
+  subroutine rayleigh_friction_init(pver, raytau0_nl, raykrange_nl, rayk0_nl, masterproc, iulog, errmsg, errflg)
+
+    integer, intent(in)                  :: pver
+    real (kind_phys), intent(in)         :: raytau0_nl
+    real (kind_phys), intent(in)         :: raykrange_nl
+    integer, intent(in)                  :: rayk0_nl
+    logical, intent(in)                  :: masterproc
+    integer, intent(in)                  :: iulog
+    character(len=512), intent(out)      :: errmsg
+    integer, intent(out)                 :: errflg
+
+    ! local variables
+    character(len=*), parameter :: subname = 'rayleigh_friction_init'
+    real (kind_phys) x
+    integer k, ierr
+
+    errmsg = ''
+    errflg = 0
+    raytau0 = raytau0_nl
+    raykrange = raykrange_nl
+    rayk0 = rayk0_nl
+
+    if ( raytau0 > 0._kind_phys) then  ! Rayleigh Friction enabled
+       ! Allocate module data
+       allocate(otau(pver), stat=ierr)
+       if (ierr /= 0) then
+          errflg = ierr
+          errmsg = trim(subname)//': Allocate of otau failed'
+          return
+       end if
+
+       !-----------------------------------------------------------------------
+       ! Compute tau array
+       !-----------------------------------------------------------------------
+
+       krange = raykrange
+       if (raykrange .eq. 0._kind_phys) krange = (rayk0 - 1) / 2._kind_phys
+
+       tau0 = (86400._kind_phys) * raytau0   ! convert to seconds
+       otau0 = 0._kind_phys
+       if (tau0 .ne. 0._kind_phys) otau0 = 1._kind_phys/tau0
+
+       do k = 1, pver
+          x = (rayk0 - k) / krange
+          otau(k) = otau0 * (1 + tanh(x)) / (2._kind_phys)
+       enddo
+    end if ! Rayleigh Friction enabled
+
+    if ( masterproc ) then
+       if (raytau0 > 0._kind_phys) then
+          write(iulog,*) 'tuning parameters rayleigh_friction_init: rayk0',rayk0
+          write(iulog,*) 'tuning parameters rayleigh_friction_init: raykrange',raykrange
+          write(iulog,*) 'tuning parameters rayleigh_friction_init: raytau0',raytau0
+       else
+          write(iulog,*) 'Rayleigh Friction not enabled.'
+       endif
+    endif
+
+  end subroutine rayleigh_friction_init
+
+  !=========================================================================================
+  !> \section arg_table_rayleigh_friction_run Argument Table
+  !! \htmlinclude rayleigh_friction_run.html
+  subroutine rayleigh_friction_run(ncol, pver, ztodt, u, v, dudt, dvdt, dsdt, errmsg, errflg)
+
+    !------------------------------Arguments--------------------------------
+    integer,             intent(in) :: ncol  
```

Comment:
I believe `ncol` is no longer needed in this routine when the changes requested are implemented

Reply:
Removed, but that opened a bag of worms